analysis_scripts/analyze_mention_v_citation/mention_v_citation.md

```diff
@@ -0,0 +1,1080 @@
+ci_df_cast = reshape2::dcast(ci_df, 
+                             year+country+address.country_code+tot_citations+tot_mentions ~ corpus, 
+                             value.var="mean")
+
+# calculate the difference between mentions + citations
+ci_df_cast$M_C = ci_df_cast$naturenews_mentions - ci_df_cast$naturenews_citations
+
+# show the spread of the difference mentions and citations
+ggplot(subset(ci_df_cast, tot_citations > MIN_ART | tot_mentions > MIN_ART), 
+       aes(x=as.numeric(year), y=as.numeric(M_C))) +
+    geom_point() + theme_bw() + 
+    geom_hline(yintercept = MIN_PROP, color="red") +
+    geom_hline(yintercept = -1*MIN_PROP, color="red") +
+    xlab("Corpus") + ylab("Mention % - Citation % for each country+year") +
+    ggtitle("Diff. btw mentions and citations for each country+year (1 point is a country)") + 
+    scale_fill_brewer(palette="Set2")
+```
+
+<img src="mention_v_citation_files/figure-markdown_github/read_in_bootstrap-3.png" style="display: block; margin: auto;" />
+
+``` r
+# final dataframe with all filters
+top_diff_MC = subset(ci_df_cast, tot_citations > MIN_ART | tot_mentions > MIN_ART)
+top_diff_MC = subset(top_diff_MC, M_C > MIN_PROP | M_C < -1*MIN_PROP)
+```
+
+#### Plot the top country-year pairings have a large difference in citations vs mentions
+
+In the previous plot, all countries that have at least one point outside of the red lines will be considered either class M (above top red line) or class C (below bottom red line). If a country is found in both groups across the years, it is removed from consideration in later downstream processing. Countries that are in both classes are shown in the heatmap as having both blue and yellow entries across the row.
+
+``` r
+make_heatmap_res <- function(in_df, value_col){
+    plot_matr_MC = reshape2::dcast(in_df, 
+                             country ~ year, 
+                             value.var=value_col)
+    row.names(plot_matr_MC) = plot_matr_MC$country
+    plot_matr_MC = plot_matr_MC[,-1]
+    #plot_matr_MC[is.na(plot_matr_MC)] = 0
+    
+    max_val = max(abs(plot_matr_MC), na.rm = T)
+    breaks = c(seq(-1*max_val, max_val, by = 0.01))
+    color_pmap <- colorRampPalette(c("yellow", "white", "blue"))(length(breaks))
+
+    if(max_val > 1){
+        breaks = c(seq(1, max_val, by = 1))
+        color_pmap <- colorRampPalette(c("white", "blue"))(length(breaks))
+
+    }
+
+    res = list(plot_matr = plot_matr_MC,
+               color_pmap = color_pmap,
+               breaks = breaks)
+    return(res)
+}
+
+# plot the Top ptoportion differences
+res_MC = make_heatmap_res(top_diff_MC, value_col="M_C")
+pheatmap(res_MC$plot_matr, cluster_rows = F, 
+         cluster_cols = F, display_numbers = T, 
+         main = "Top (Mention - Citation) Proportions",
+         color = res_MC$color_pmap, breaks = res_MC$breaks)
+```
+
+<img src="mention_v_citation_files/figure-markdown_github/plot_filtered_countries-1.png" style="display: block; margin: auto;" />
```

Comment:
So I only include countries in the plot where there is a significant amount of either M>C or M<C. Everywhere that is white could be blue or yellow, but since its not significant I don't show it. Does that make it clearer?